typescript/packages/subsurface-viewer/src/components/Map.tsx

```diff
@@ -1634,11 +1713,71 @@ function applyZScale(
     target: Point2D | Point3D | undefined,
     zScale: number
 ): void {
-    if (target?.[2]) {
+    if (target?.[2] != undefined) {
         target[2] = target[2] * zScale;
     }
 }
 
+/**
+ * Returns a z-scaled target.
+ * This is needed, as the camera target is specified in world coordinates, while the Z scale
+ * is applied to the transformation matrix of the object coordinates. The target must be applied
+ * the same scale to be consistent with the display.
+ * @param target camera target that must take into account the Z scale.
+ * @param zScale Z scale.
+ */
+function zScaledTarget(
+    target: Point2D | Point3D | undefined,
+    zScale: number
+): Point2D | Point3D | undefined {
+    if (!target) {
+        return undefined;
+    }
+    if (target[2] == undefined) {
+        return [target[0], target[1]];
+    }
+
+    return [target[0], target[1], target[2] * zScale];
+}
+
+/**
+ * Returns an inverted z-scaled target.
+ * This is needed, as the camera target is specified in world coordinates, while the Z scale
+ * is applied to the transformation matrix of the object coordinates. The target must be applied
+ * the same scale to be consistent with the display.
+ * @param target camera scaled target.
+ * @param zScale Z scale.
+ * @param unscaledTarget last known unscaled target which can be taken as a fallback.
+ */
+function inversedZScaled(
+    target: Point2D | Point3D | undefined,
+    zScale: number,
+    unscaledTarget?: Point2D | Point3D | undefined
+): Point2D | Point3D | undefined {
+    if (!target) {
+        return undefined;
+    }
+    if (target[2] == undefined) {
+        return [target[0], target[1]];
+    }
+    if (zScale != 0) {
+        return [target[0], target[1], target[2] / zScale];
+    }
+
+    if (
+        unscaledTarget?.[2] != undefined &&
+        target[0] === unscaledTarget?.[0] &&
+        target[1] === unscaledTarget?.[1]
+    ) {
+        return [
+            target[0],
+            target[1],
+            target[2] ? target[2] : unscaledTarget[2],
+        ];
+    }
+    return [target[0], target[1], target[2]];
+}
```

Comment:
Would be good with some unit tests. YCR

Reply:
It will be a good try for copilot.
I'll do it in a separate PR.